lib/runtime/storage/storagediff.go

```diff
@@ -290,3 +296,27 @@ func (cs *storageDiff) applyToTrie(t trie.Trie) {
 
 	}
 }
+
+func (cs *storageDiff) insertSortedKey(key string) {
+	pos := sort.Search(len(cs.sortedKeys), func(i int) bool {
+		return cs.sortedKeys[i] >= key
+	})
+
+	if pos < len(cs.sortedKeys) && cs.sortedKeys[pos] == key {
+		return // key already exists
+	}
+
+	cs.sortedKeys = append(cs.sortedKeys, "")
+	copy(cs.sortedKeys[pos+1:], cs.sortedKeys[pos:])
+	cs.sortedKeys[pos] = key
+}
+
+func (cs *storageDiff) removeSortedKey(key string) {
+	pos := sort.Search(len(cs.sortedKeys), func(i int) bool {
+		return cs.sortedKeys[i] >= key
+	})
```

Comment:
Given that the list is sorted can we use slice.BinarySearch https://pkg.go.dev/slices#BinarySearch